### PR TITLE
Profile picture2

### PR DESCRIPTION
**Pull request by aLeuleu**
Screenshot 2024-02-10 at 22 58 21

GET /user/avatar/<username> : get the avatar (no auth needed)
POST /user/avatar/<username> : post avatar ( < 1Mb, auth required)
DELETE /user/avatar/<username> : rm avatar and replace with default avatar (auth required)

### CODE REVIEW

**Review by EthanDelage — APPROVED** [review]

File: user_management/src/user/views/avatar.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+from user_management.settings import MEDIA_ROOT, STATIC_ROOT
+from user_management.utils import save_image_from_base64
+
+
+@method_decorator(user_authentication(['POST', 'GET', 'DELETE']), name='dispatch')
+@method_decorator(csrf_exempt, name='dispatch')
+class AvatarView(View):
+    @staticmethod
+    def get(request, username):
+        user = User.objects.filter(username=username).first()
```
Comment (hferraud) [inline user_management/src/user/views/avatar.py:21]: try this line

File: user_management/src/user/views/avatar.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        return FileResponse(open(path, 'rb'), content_type='image/png')
+
+    @staticmethod
+    def post(request, username):
+        try:
+            json_request = json.loads(request.body.decode('utf-8'))
+        except Exception as e:
+            return JsonResponse(data={'error': f'Invalid JSON : {e}'}, status=400)
+        user_id = get_user_id(request)
+        user = User.objects.filter(id=user_id).first()
```
Comment (hferraud) [inline user_management/src/user/views/avatar.py:40]: try this line

File: user_management/src/user/views/avatar.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        base64_string = base64_string.replace('data:image/png;base64,', '')
+        success, error = save_image_from_base64(base64_string, user)
+        if not success:
+            return JsonResponse(data={'error': error}, status=400)
+        return JsonResponse(data={'message': 'Avatar updated'}, status=200)
+
+    @staticmethod
+    def delete(request, username):
+        user_id = get_user_id(request)
+        user = User.objects.filter(id=user_id).first()
```
Comment (hferraud) [inline user_management/src/user/views/avatar.py:62]: try this line

**Review by hferraud — CHANGES_REQUESTED** [review]

**Review by hferraud — APPROVED** [review]

**Review by tdameros — CHANGES_REQUESTED** [review]
Please can you disable authentication on the endpoint get profile images because I'm going to use them on HTML  tags that won't be able to authenticate. Thanks a lot!

**Comment by tdameros** [comment]
Can you make sure that POST images aren't too big so that a client can't overload the server with huge images?
For example, a limit of 2MB per image would be correct.

File: user_management/doc/User_management.md [hunk 4]
```diff
@@ -652,3 +652,79 @@ all fields are mandatory
...
+### Get user's avatar
+
+<details>
+ <summary><code>GET</code><code><b>/user/avatar/</b></code></summary>
+
+### Parameters
+
+Authorization: {access_token}
+
+NB : any access_token can be used to get the avatar of any user
```
Comment (tdameros) [inline user_management/doc/User_management.md:669]: The documentation should be updated on this point, as it is no longer the case.

**Review by tdameros — CHANGES_REQUESTED** [review]

File: user_management/doc/User_management.md (hunk 4 above)
Comment (aLeuleu) [reply user_management/doc/User_management.md]: [400a648](https://github.com/tdameros/42-transcendence/pull/210/commits/400a648f8a0c55b937561a18e61d914d62266666)

**Review by tdameros — APPROVED** [review]

**Pull request merged**